data/src/main/java/me/androidbox/data/worker_manager/TaskCreateSynchronizerWorker.kt

```diff
@@ -0,0 +1,150 @@
+package me.androidbox.data.worker_manager
+
+import android.content.Context
+import androidx.hilt.work.HiltWorker
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.supervisorScope
+import me.androidbox.data.local.dao.ReminderDao
+import me.androidbox.data.local.dao.TaskDao
+import me.androidbox.data.local.entity.ReminderEntity
+import me.androidbox.data.local.entity.TaskEntity
+import me.androidbox.data.mapper.toReminderDto
+import me.androidbox.data.mapper.toTaskDto
+import me.androidbox.data.remote.network.reminder.ReminderService
+import me.androidbox.data.remote.network.task.TaskService
+import me.androidbox.data.remote.util.CheckResult.checkResult
+import me.androidbox.data.worker_manager.constant.Constant.RETRY_COUNT
+import me.androidbox.domain.constant.SyncAgendaType
+
+@HiltWorker
+/** TODO Rename to something more readable as this is not just related to tasks */
+class TaskCreateSynchronizerWorker @AssistedInject constructor(
+    @Assisted private val context: Context,
+    @Assisted private val workerParameters: WorkerParameters,
+    private val taskService: TaskService,
+    private val taskDao: TaskDao,
+    private val reminderService: ReminderService,
+    private val reminderDao: ReminderDao,
+) : CoroutineWorker(context, workerParameters) {
+
+    override suspend fun doWork(): Result {
+
+        /** TODO is this the correct place to have this check and to retry or fail */
+        if(runAttemptCount > RETRY_COUNT) {
+            Result.failure()
+        }
+        else {
+            Result.retry()
+        }
+
+        val createdTasks: Deferred<List<TaskEntity>>
+        val updatedTasks: Deferred<List<TaskEntity>>
+        val createdReminders: Deferred<List<ReminderEntity>>
+        val updatedReminders: Deferred<List<ReminderEntity>>
+
+        /** Get all the created, updated tasks and reminders
+         *  TODO As these will run sequentially - is it better to use supervisor scope and run them in parallel? */
```

Comment:
Yea that's fine, tho what I like to do is to not have "Deferred<List", and instead use actual lists and just .awaitAll()